<?xml version="1.0"?>
<launch>
    <arg name="range" default="18.0"/>
    <arg name="grid_num" default="80"/>
    <arg name="hz" default="100.0"/>
    <arg name="crop" default="10"/>
    <arg name="is_save_image" default="false"/>
    <arg name="save_data_num" default="1"/>
    <arg name="child_frame_id" default="base_footprint"/>
	<arg name="pkg_path" default="/home/amsl/ros_catkin_ws/src/bev_converter/bev_img"/>

	<param name="use_sim_time" value="false"/>

	<node pkg="bev_converter" type="bev_converter" name="bev_converter">
		<param name="RANGE" value="$(arg range)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="Hz" value="$(arg hz)" type="double"/>
		<param name="CHILD_FRAME_ID" value="$(arg child_frame_id)" type="str"/>
	</node>

	<node pkg="bev_converter" type="bev_flow_estimator" name="bev_flow_estimator">
		<param name="RANGE" value="$(arg range)" type="double"/>
		<param name="GRID_NUM" value="$(arg grid_num)" type="int"/>
		<param name="Hz" value="$(arg hz)" type="double"/>
		<param name="MANUAL_CROP_SIZE" value="$(arg crop)" type="int"/>
		<param name="IS_SAVE_IMAGE" value="$(arg is_save_image)" type="bool"/>
		<param name="SAVE_NUMBER" value="$(arg save_data_num)" type="int"/>
        <!-- <rosparam command="load" file="$(find cats)/config/cats_motion_param.yaml"/> -->
	</node>

	<node pkg="bev_converter" type="pointcloud_converter" name="pointcloud_converter">
	</node>
	<node pkg="bev_converter" type="odom_publisher" name="odom_publisher">
	</node>

    <arg name="width" default="40.0"/>
    <arg name="resolution" default="0.2"/>
    <arg name="occupancy_threshold" default="0.2"/>
    <arg name="beam_num" default="720"/>
    <arg name="skip_scan_flag" default="false"/>

    <node pkg="dynamic_cloud_detector" type="dynamic_cloud_detector" name="dynamic_cloud_detector">
        <param name="WIDTH" value="$(arg width)"/>
        <param name="RESOLUTION" value="$(arg resolution)"/>
        <param name="OCCUPANCY_THRESHOLD" value="$(arg occupancy_threshold)"/>
        <param name="BEAM_NUM" value="$(arg beam_num)"/>
        <param name="SKIP_SCAN_FLAG" value="$(arg skip_scan_flag)"/>
        <remap from="/velodyne_obstacles" to="/pedsim_people_sensor/point_cloud2_local"/>
        <remap from="/odom/complement" to="/odom"/>
        <!-- <remap from="/odom/complement" to="/bev_odom"/> -->
    </node>
	<!-- optional -->
	<!-- <include file="$(find bev_converter)/launch/bag_player.launch"/> -->
</launch>
